x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/WeightedTokensQueryBuilder.java

```diff
@@ -152,27 +153,57 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
         if (ft == null) {
             return new MatchNoDocsQuery("The \"" + getName() + "\" query is against a field that does not exist");
         }
+
+        final String fieldTypeName = ft.typeName();
+        if (AllowedFieldType.isFieldTypeAllowed(fieldTypeName) == false) {
+            throw new ElasticsearchParseException(
+                "["
+                    + fieldTypeName
+                    + "]"
+                    + " is not an appropriate field type for text expansion query. "
```

Comment:
Technically this error can be thrown for a text expansion query OR a weighted tokens query. 

```suggestion
                    + " is not an appropriate field type for this query. "
```

Reply:
Updated